### PR TITLE
[x86/Linux] Fix EnclosingClauseCallerSP

### PR DESCRIPTION
**Pull request by wateret**
Seems like this is missing part of x86/linux port.

### CODE REVIEW

**Comment by wateret** [comment]
Test results on x86

version : master w/ #9601
```
=======================
     Test Results
=======================
# CoreCLR Bin Dir  :
# Tests Discovered : 7034
# Passed           : 6287
# Failed           : 469
# Skipped          : 278
=======================
```

version : master w/ #9601 and this PR
```
=======================
     Test Results
=======================
# CoreCLR Bin Dir  :
# Tests Discovered : 7034
# Passed           : 6308
# Failed           : 448
# Skipped          : 278
=======================
```

File: src/vm/exceptionhandling.cpp [hunk 1]
```diff
@@ -1930,12 +1930,12 @@ CLRUnwindStatus ExceptionTracker::ProcessOSExceptionNotification(
...
             UINT_PTR EnclosingClauseCallerSP = (UINT_PTR)pDispatcherContext->EstablisherFrame;
-#elif defined(_TARGET_AMD64_)            
-            // Extract the CallerSP from RegDisplay on AMD64
+#elif defined(_TARGET_AMD64_) || defined(_TARGET_X86_)
+            // Extract the CallerSP from RegDisplay on AMD64 and x86
             REGDISPLAY *pRD = cfThisFrame.GetRegisterSet();
             _ASSERTE(pRD->IsCallerContextValid || pRD->IsCallerSPValid);
             UINT_PTR EnclosingClauseCallerSP = (UINT_PTR)GetSP(pRD->pCallerContext);
-#else // !_ARM_ && !_AMD64_ && !_ARM64_
+#else // !_ARM_ && !_AMD64_ && !_ARM64_ && !_X86_
```
Comment (janvorli) [inline src/vm/exceptionhandling.cpp]: A nit - can you please fix the comment to reflect the real define names (`_TARGET_XXX_`)?

**Comment by wateret** [comment]
cc: @parjong @jkotas @janvorli

File: src/vm/exceptionhandling.cpp [hunk 2]
```diff
@@ -1930,12 +1930,12 @@ CLRUnwindStatus ExceptionTracker::ProcessOSExceptionNotification(
             // DispatcherContext->EstablisherFrame's value
             // represents the CallerSP of the current frame.
             UINT_PTR EnclosingClauseCallerSP = (UINT_PTR)pDispatcherContext->EstablisherFrame;
-#elif defined(_TARGET_AMD64_)            
-            // Extract the CallerSP from RegDisplay on AMD64
+#elif defined(_TARGET_AMD64_) || defined(_TARGET_X86_)
```
Comment (janvorli) [inline src/vm/exceptionhandling.cpp]: I am not sure if we want to use the ARM or AMD64 way for x86 - did we already make a choice in this matter? I 
If we follow the amd64 way, then we also need to fix the code here:
https://github.com/dotnet/coreclr/blob/master/src/vm/exceptionhandling.cpp#L2843-L2849
and here:
https://github.com/dotnet/coreclr/blob/master/src/vm/exceptionhandling.cpp#L3109-L3115
(and if not, we should modify the comments at those places to mention x86).

**Comment by stephentoub** [comment]
@dotnet-bot test this please (Jenkins was down)

**Comment by wateret** [comment]
Closing since #9814 handles this.

**Pull request closed without merging**